<launch>
  <arg name="use_rp" default="true"/>
  <arg name="use_velo" default="true"/>
  <arg name="use_joy" default="true"/>
  <arg name="map_file" default="/xavier_ssd/leo_ws/map.yaml"/>

  <include file="$(find leo_navigation)/launch/amcl.launch">
    <arg name="map_file" value="$(arg map_file)"/>
  </include> 
  <include if="$(arg use_joy)" file="$(find teleop_twist_joy)/launch/teleop.launch"/>
  <include if="$(arg use_velo)" file="$(find velodyne_pointcloud)/launch/VLP16_points.launch"/>
  <include file="$(find leo_navigation)/launch/odometry.launch"/>
  
  <node if="$(arg use_rp)" name="rplidarNode_front"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
    <param name="serial_port"         type="string" value="/dev/rplidar"/>
    <param name="serial_baudrate"     type="int"    value="115200"/><!--A1/A2 -->
    <!--param name="serial_baudrate"     type="int"    value="256000"--><!--A3 -->
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
    <param name="output_scan"         type="string" value="/scan"/>
  </node>

  <node pkg="leo_driving" type="leo_driving_node" name="leo_driving_node" output="screen">
	<param name="Kpy_param" value="0.1"  /> 
	<param name="Kpy_param_rot" value="0.01"  /> 
	<param name="linear_vel" value="0.0" /> 
	<param name="linear_vel_rot" value="0.01" /> 
	<param name="boundary_percent" value="0.02" /> 
	<param name="side_obstacle_dist" value="0.2" /> 
	<param name="front_obstacle_dist" value="0.5" /> 
	<param name="obastalce_coefficient" value="0.1" /> 
	<param name="rotation_ang_err" value="0.2" /> 
	<param name="rotation_dist_err_" value="0.2" />  
	<param name="back_rotating_ang_" value="3.0" /> 
  </node>

<node pkg="tf" type="static_transform_publisher" name="base_link_to_rp" args="0 0 0 3.141592 0 0 base_link laser 100" />
<node pkg="tf" type="static_transform_publisher" name="base_link_to_velo" args="0 0 0 0 0 0 base_link velodyne 100" />
<node pkg="tf" type="static_transform_publisher" name="base_link_to_imu" args="0 0 0 0.0 0 0 base_link imu 100" />
</launch>

